system_tests/queue_system_test.go

```diff
@@ -101,4 +102,44 @@ var _ = Describe("Queue Controller", func() {
 		}, 30).Should(HaveOccurred())
 		Expect(err.Error()).To(ContainSubstring("Object Not Found"))
 	})
+
+	It("with DeletionPolicy=retain, deletes k8s resource but keeps the queue in RabbitMQ", func() {
```

Comment:
Same as with Federation system tests. I think it's better to surround this `It` with `When` like `When("deletion policy is retain", func() {` because it reads out more naturally.